.pre-commit-config.yaml

```
@@ -0,0 +1,40 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+---
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v2.4.0
+    hooks:
+      - id: check-added-large-files
+      - id: check-ast
+      - id: check-case-conflict
+      - id: check-merge-conflict
+      - id: debug-statements
+      - id: end-of-file-fixer
+      - id: trailing-whitespace
+        args: [--markdown-linebreak-ext=md]
+  - repo: https://github.com/adrienverge/yamllint
+    rev: ''
+    hooks:
+      - id: yamllint
+  - repo: https://github.com/codespell-project/codespell
+    rev: ''
+    hooks:
+      - id: codespell
+        exclude: 'esmvaltool/diag_scripts/cvdp'
```

```suggestion
        exclude: 'esmvalcore/cmor/tables'
```
and the other linters should also be disabled for this directory. Would it be possible to make a single regex with paths to ignore and then add that with a yaml anchor to all checkers?

---

Apparently there is a top level exclude which we can use: https://pre-commit.com/#top_level-exclude

It can be formatted as ar regex, which other folders should be excluded?

---

Very good, preferably anything in the folders
- esmvalcore/cmor/tables
- esmvalcore/preprocessor/ne_masks